src/libraries/System.Diagnostics.Tracing/tests/TrimmingTests/EventSourceManifestTest.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System;
+
+/// <summary>
+/// Tests that using an EventSource to get the manifest works without method references in a trimmed application.
+/// EventSource has DynamicallyAccessedMembersAttribute applied at class level which means derived types keeps members
+/// </summary>
+internal class Program
+{
+    internal class EventSourceTest2 : EventSource
+    {
+        public void EventSourceTest2_Method_0() => WriteEvent(1); 
+
+        public void EventSourceTest2_Method_1(int value) => WriteEvent(2, value); 
+        void EventSourceTest2_Method_2(string name) => WriteEvent(3, name); 
+
+        [NonEvent]
+        public void EventSourceTest2_Method_3(int num1, int num2) => WriteEvent(4, num1, num2); 
+        [NonEvent]
+        void EventSourceTest2_Method_4(){}
+
+        [Event(500)]
+        public void EventSourceTest2_Method_5(byte[] bytes) => WriteEvent(500, bytes); 
+        [Event(1500)]
+        protected virtual void EventSourceTest2_Method_6(long value) => WriteEvent(1500, value); 
+
+        [Event(2500)]
+        int EventSourceTest2_Method_7() => 5; 
+    }
+
+    public static int Main()
+    {
+        string manifest = EventSource.GenerateManifest(typeof(EventSourceTest2), null);
+        // we are going to avoid as much as possible program constructs that could give the trimmer reasons to keep members
+        int testReturnValue = 100;
+        const string baseMethodName = "EventSourceTest2_Method_";
+        int[] exclusions = { 3, 4, 8 };
+        for (int i = 0; i <= 8; i++)
+        {
+            string methodName = $"{baseMethodName}{i}";
+            // We expect the methodName to be in the manifest unless the prefix, i, is in exclusions ([NonEvent] or non-existing methods)
+            if (manifest.Contains(methodName) ? exclusions.Contains(i) : !exclusions.Contains(i))
+                testReturnValue = -1;
```

Comment:
I wonder if it would be easier to read like the following:

```suggestion
bool methodExists = manifest.Contains(methodName);
bool shouldMethodExist = !exclusions.Contains(i);

if (methodExists != shouldMethodExist)
{
    return -1;
}
```